### DISCUSSION TITLE
I'm trying to create a Spanish TTS Model

### DISCUSSION THREAD

**Discussion by BossyFi** [coqui-ai/TTS#1617] [General Q&A]
Hello, I discovered this world some months ago and since then I'm trying to create a TTS Model for a college project. I started with the Coqui recipes and trained a Glow Model. This model was trained for 50k steps with 3 hours of corpus, but didn't sound very well. 

https://user-images.githubusercontent.com/61434659/171877534-fda4295d-6086-480b-a38b-0862ee4c68dd.mp4

I continued reading and learned that maybe, the vocoder was the problem, so I also trained a vocoder, in this case an Univnet Vocoder. The results were way better. With only 10k steps, I had this result.

https://user-images.githubusercontent.com/61434659/171877864-a8583098-ef53-4680-9db3-35d9975747d3.mp4

The model was trained for another 20K steps (I plan to continue training the vocoder for some steps more) but it didn't look like it was getting better. 

So in the meantime, I thought about training a VITS model, thinking that maybe, an end to end model could give me decent results with a small corpus. But now I'm getting this error when reaching ~4000 steps
![image](https://user-images.githubusercontent.com/61434659/171878610-6e673d12-4737-4ce3-9786-dcbd26491d2a.png)

Could someone give me some advice for the training? Should I keep training my Glow + Univnet Model for some steps more? Right now I'm in a spot where I kinda understad but still have a lot of doubts. Thanks in advance.

**Comment by neurlang** [comment]
Hello sir. Please kindly fix the `FirstVITSModel-June-03-2022_10+40AM-0000000/config.json`
Inside, there is a test sequences section:
```
    "test_sentences": [
        [
            "It took me quite a long time to develop a voice, and now that I have it I'm not going to be silent."
        ],
        [
            "Be a voice, not an echo."
        ],
        [
            "I'm sorry Dave. I'm afraid I can't do that."
        ],
        [
            "This cake is great. It's so delicious and moist."
        ],
        [
            "Prior to November 22, 1963."
        ]
    ],
```
Please delete all the test sequences (they are english and are not needed for spanish):
```
    "test_sentences": [    ],
```
retrain again, problem will not longer manifest.

**Reply by BossyFi** [reply]
Thank you so much, I'll try deleting the test sentences!

**Comment by iprovalo** [comment]
@BossyFi do you mind sharing your glow_tts config for Spanish language?  I am trying to train based on ljspeech dataset (female/karen_savage/angelina).  While the vocoder (mb melgan) is training fine, glow-tts is not working.

Glow TTS:
<img width="916" alt="Screen Shot 2022-07-15 at 5 52 07 PM" src="https://user-images.githubusercontent.com/1660997/179260719-86ac4cb4-7800-4346-9bd8-9138f4ae94be.png">


Config: [config.json.txt](https://github.com/coqui-ai/TTS/files/9124622/config.json.txt)

**Reply by BossyFi** [reply]
Of course, this is the one I used back then
[config.txt](https://github.com/coqui-ai/TTS/files/9125629/config.txt)

**Reply by iprovalo** [reply]
Thank you, @BossyFi !

Which dataset did you use for this?

**Reply by BossyFi** [reply]
You're welcome!
I made my own dataset.

**Reply by BossyFi** [reply]
I recommend you to check this video https://www.youtube.com/watch?v=C62nykAda7w and also the channel. It has some good explanations about Coqui.

**Reply by iprovalo** [reply]
Thank you, @BossyFi !

**Comment by smartos99** [comment]
Hello! Can you give a short tutorial on how did you train your model in Spanish? The documentation is too broad and there are aspects I can't see clearly. For example as you used your own dataset, did you have to create your own formatter and text_cleaner, because I didn't find a Spanish text cleaner in the coqui/tts documentation? Any help would be very much appreciated

**Reply by BossyFi** [reply]
Hi @smartos99, I was working with Coqui.ai months ago but couldn't achieve good results with my small dataset (3 hours) so I tried others repositories (I'm sorry if talking about others repositories is not allowed) and discovered [FakeYou](https://fakeyou.com/). They have very good material to train models directly in Spanish using Tacotron 2.

You can find all the material in their discord. 

Just today, I presented my Final Degree Project about this matter and will be up soon. I'll post a link to the paper if you're interested. It's written in Spanish, but I'm assuming that you also speak Spanish so that won't be a problem.

**Reply by smartos99** [reply]
Thank you very much! 
Yes please, a link to the paper would be very much appreciated.

**Reply by BossyFi** [reply]
I hope is helpful [Búsqueda de voz natural para habla sintética mediante el uso de redes neuronales](https://www.linkedin.com/posts/bozhidar-petrov-valchev-491790222_b%C3%BAsqueda-de-voz-natural-para-habla-sint%C3%A9tica-activity-6978237395401879552-LDnN?utm_source=share&utm_medium=member_desktop)
